src/Bicep.Core/Registry/Providers/TypesV1Archive.cs

```diff
@@ -44,4 +49,21 @@ private static async Task AddFileToTar(TarWriter tarWriter, string archivePath,
 
         await tarWriter.WriteEntryAsync(tarEntry);
     }
+
+    private static HashSet<string> getAllUniqueTypePaths(string pathToIndex)
+    {
+        FileStream indexStream = new(pathToIndex, FileMode.Open, FileAccess.Read);
```

Comment:
Use the `IFileSystem` equivalent, because it can be mocked in tests.

```suggestion
        var indexStream = fileSystem.FileStream.New(pathToIndex, FileMode.Open, FileAccess.Read);
```